SharpPcap/ICaptureDevice.cs

```diff
@@ -166,6 +166,14 @@ public interface ICaptureDevice : IDisposable
         /// Return the pcap link layer value of an adapter. 
         /// </summary>
         PacketDotNet.LinkLayers LinkType { get; }
+
+        #region Timestamp
+        /// <summary>
+        /// Note: It isn't possible to set the precision on a device that has already
+        /// been activated. Please Open() the device with the desired precision.
+        /// </summary>
+        TimestampResolution TimestampPrecision { get; }
```

Comment:
`Precision` or `Resolution` ?

Reply:
it should be available for all pcap devices, so live, file reader and writer.

You noticed the precision vs. resolution usage too? It's all of the place in libpcap docs and code.

So the function uses precision, https://github.com/the-tcpdump-group/libpcap/blob/master/pcap/pcap.h#L428

but they do use resolution in some language. The variables are called precision though so let me rename appropriately. It's confusing inside of their documentation for sure. And language wise they are not the same thing. I think 'resolution' is the correct term though as this is the smallest unit vs. precision which is how close the values are to the actual value.

If we standardize on what libpcap uses that would mean using 'precision'. Properly I think we should use 'resolution' a s its the correct term.

Reply:
@chmorgan there one detail I could clarify:
* resolution set by `PCAP_TSTAMP_PRECISION_NANO` affects how the timestamp `timeval` should be parsed (the microsecond field becomes a nanosecond field) 
* All other timestamp api affects how libpcap gets the timestamp but not how you should parse it.

Reply:
@chmorgan Now I think about it, since `pcap_set_tstamp_precision` changes how the timestamp value should be parsed.
It would be better to split this PR into two smaller ones:
* one for timestamp type `pcap_set_tstamp_type` and related functions
* one for timestamp resolution `pcap_set_tstamp_precision` and related functions, and updating `PosixTimeval`

Reply:
Yep. Well its more about how the user would parse it. My thought, even though it isn't implemented yet, is to have a flag on PosixTimeval to indicate which resolution it has and expose a tuple of either (second, microsecond) or (second, nanosecond) that can be retrieved. We could have separate types for each, like PosixTimevalMS or PosixTimevalNS but I wasn't sure that would help the user as much.